### PR TITLE
Synchronise teams on GitHub with the state of teams in the hub

### PR DESCRIPTION
**Pull request by jits**
**Important design decision:**

Teams on GitHub are created – and subsequently retrieved and matched – using the form `[Hub] {team.slug}` (e.g. `[Hub] operations-team`) as the **name** of the team on GitHub. We do this – instead of relying on the GitHub team ID – in order to provide idempotent operations, and to more easily recover from partial failures.

**Highlights:**

- Lock down the connecting of an identity flow – only allow connecting an identity if the user still has access to that integration (via any of their teams).
- Handle the following "events" in the system and ensure that the state of teams on GitHub matches that of the hub:
  - Team created
  - Team deleted
  - Team membership created or updated
  - Team membership deleted
  - Identity connected (note: identities are tied to a particular integration)
  - Identity disconnected
  - Integration created or updated (since allocations _may_ have been added/removed)
- … the system above can now be easily extended for any other provider.
- Creating a code repo now assigns the relevant team to that repo on GitHub.
- New queue (`teams`) for processing these team sync tasks using FIFO – see `README.md` for details.
- End-to-end integration spec to test out as many cases as possible.
- BREAKING: Remove the `all_team_id` config from the `git_hub` provider integration – we no longer assign this "global" team to repos, given the new team specific GitHub teams.
- BREAKING: Had to remove the option to delete an integration in the hub UI – instead we need to (later) implement a proper async deletion _request_ for integrations, which would go through and remove allocations (and thus handle deletion of teams upstream at the providers), request deletion for resources, and carry out any other proper clean up work before finally deleting the integration entry in the db (captured in https://github.com/appvia/appvia-hub/issues/98).

### CODE REVIEW

**Comment by danielwhatmuff** [comment]
LG. Obvs needs thorough testing once deployed which Im happy to help with.

File: README.md [hunk 1]
```diff
@@ -97,10 +97,11 @@ There are different background workers for different types of jobs:
 - One for **resource provisioning** specifically:
   - **IMPORTANT:** MUST run with a concurrency of 1 (`-c 1`) to ensure proper FIFO processing.
   - `bundle exec sidekiq -q resources -c 1`
+- One for **teams** processing specifically:
+  - **IMPORTANT:** MUST run with a concurrency of 1 (`-c 1`) to ensure proper FIFO processing.
+  - `bundle exec sidekiq -q teams -c 1`
```
Comment (davet1985) [inline README.md]: I'm guessing this is going to require changes to the environments we have running?

Comment (jits) [reply README.md]: That's correct – sorry 🙂

Comment (davet1985) [reply README.md]: No worries, just need to keep track of what's required 😄

**Pull request merged**